src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -227,51 +227,30 @@ Copyright (c) .NET Foundation. All rights reserved.
   <UsingTask TaskName="ResolvePublishAssemblies" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
   <Target Name="RunResolvePublishAssemblies"
           DependsOnTargets="ComputePrivateAssetsPackageReferences;
-                            ComputePackagesToBeFiltered;
                             _DefaultMicrosoftNETPlatformLibrary">
-
+    <ItemGroup>
+      <FilterProjectFileList Include="$(FilterProjectFiles.Split('%3B'))"/>
+    </ItemGroup>
     <ResolvePublishAssemblies ProjectPath="$(MSBuildProjectFullPath)"
                               AssetsFilePath="$(ProjectAssetsFile)"
-                              FilterProjectAssetsFile="$(FilterProjectAssetsFile)"
                               TargetFramework="$(TargetFrameworkMoniker)"
                               RuntimeIdentifier="$(RuntimeIdentifier)"
                               PlatformLibraryName="$(MicrosoftNETPlatformLibrary)"
                               PrivateAssetsPackageReferences="@(PrivateAssetsPackageReference)"
+                              FilterProjectFiles="@(FilterProjectFileList)"
                               PreserveCacheLayout="$(PreserveCacheLayout)">
 
       <Output TaskParameter="AssembliesToPublish" ItemName="ResolvedAssembliesToPublish" />
-
+      <Output TaskParameter="PackagesResolved" ItemName="PackagesThatWhereResolved" />
```

Comment:
I am aggregating all parallel computations of PackagesResolved in  AllResolvedPackagesPublished